core/types/arbitrum_legacy_tx.go

```diff
@@ -13,9 +14,25 @@ type ArbitrumLegacyTxData struct {
 	HashOverride      common.Hash // Hash cannot be locally computed from other fields
 	EffectiveGasPrice uint64
 	L1BlockNumber     uint64
+	Sender            *common.Address // only used in unsigned Txs
 }
 
-func NewArbitrumLegacyTx(origTx *Transaction, hashOverride common.Hash, effectiveGas uint64, l1Block uint64) (*Transaction, error) {
+type arbitrumLegacyTxDataNoSenderRLP struct {
+	LegacyTx
+	HashOverride      common.Hash // Hash cannot be locally computed from other fields
+	EffectiveGasPrice uint64
+	L1BlockNumber     uint64
+}
+
+type arbitrumLegacyTxDataWithSenderRLP struct {
+	LegacyTx
+	HashOverride      common.Hash // Hash cannot be locally computed from other fields
+	EffectiveGasPrice uint64
+	L1BlockNumber     uint64
+	Sender            common.Address // only used in unsigned Txs
```

Comment:
What I'd recommend doing here instead of having two structs is adding a new field which looks like this:
```go
Sender *common.Address `rlp:"optional"`
```
This should let it deserialize transaction data with or without a sender, without using two different structs.

Reply:
nice! didn't know about that one, Fixed.